python/plugins/processing/core/parameters.py

```diff
@@ -79,13 +80,25 @@ def setValue(self, obj):
         of parameter.
         """
         if obj is None:
-            if not self.optional:
+            if not self.optional and self.default is None:
                 return False
-            self.value = None
-            return True
+            elif self.default is None:
+                self.value = None
+                return True
+            else:
```

Comment:
Thanks @volaya

setDefaultValue has to be used explicitly ?

```python
param = Parameter(optional=True)
assertEqual(param.value, None)
param.default = 'foo'
assert(param.setDefaultValue())
assertEqual(param.value, "foo")
assert(param.setValue(None))
assertEqual(param.value, None)
```
Are you agree with this test ?